R/TealAppDriver.R

```diff
@@ -0,0 +1,410 @@
+    #'
+    #' @return A function similar to [shiny::NS()] that is used to create a `character`
+    #' namespace for the shiny UI.
+    #'
+    helper_NS = function(namespace, ...) { # nolint: object_name_linter.
+      dots <- rlang::list2(...)
+      checkmate::assert_list(dots, types = "character")
+      base_id <- namespace
+      if (length(dots) > 0) base_id <- paste(c(namespace, dots), collapse = shiny::ns.sep)
+
+      function(..., .css_prefix = "", .css_suffix = "") {
+        dots <- rlang::list2(...)
+        checkmate::assert_list(dots, types = "character")
+        base_string <- sprintf("%s%s%s", .css_prefix, base_id, .css_suffix)
+        if (length(dots) == 0) {
+          return(base_string)
+        }
+        (shiny::NS(base_string))(paste(dots, collapse = shiny::ns.sep))
+      }
+    },
+    #' @description
+    #' Get the input from the module in the `teal` app.
+    #' This function will only access inputs from the name space of the current active teal module.
+    #'
+    #' @param input_id (character) The shiny input id to get the value from.
+    #'
+    #' @return The value of the shiny input.
+    get_module_input = function(input_id) {
+      self$app$get_value(input = sprintf("%s-%s", self$get_active_ns("module"), input_id))
+    },
+    #' @description
+    #' Get the output from the module in the `teal` app.
+    #' This function will only access outputs from the name space of the current active teal module.
+    #'
+    #' @param output_id (character) The shiny output id to get the value from.
+    #'
+    #' @return The value of the shiny output.
+    get_module_output = function(output_id) {
+      self$app$get_value(output = sprintf("%s-%s", self$get_active_ns("module"), output_id))
+    },
+    #' @description
+    #' Set the input in the module in the `teal` app.
+    #' This function will only set inputs in the name space of the current active teal module.
+    #'
+    #' @param input_id (character) The shiny input id to get the value from.
+    #' @param value The value to set the input to.
+    #'
+    #' @return The `TealAppDriver` object invisibly.
+    set_module_input = function(input_id, value) {
+      self$app$set_inputs(
+        !!sprintf("%s-%s", self$get_active_ns("module"), input_id) := value
+      )
+      invisible(self$app)
+    },
+    #' @description
+    #' Get the active datasets that can be accessed via the filter panel of the current active teal module.
+    get_active_filter_vars = function() {
+      displayed_datasets_index <- sapply(
+        self$app$get_html(
+          sprintf(
+            "#%s-active-filter_active_vars_contents > span",
+            self$get_active_ns("filter_panel")
+          )
+        ),
+        function(x) {
+          style <- x %>%
+            rvest::read_html() %>%
+            rvest::html_node("span") %>%
+            rvest::html_attr("style")
+          style <- ifelse(is.na(style), "", style)
```

Comment:
Please don't use `ifelse` like this unless it is absolutely impossible for `style` to be of `length != 1`.